### DISCUSSION TITLE
数字开头的句子无法识别

### DISCUSSION THREAD

**Discussion by joseph16388** [SWHL/RapidVideOCR#69] [Q&A]
RapidVideOCR和onnxruntime都是最新版本

![image](https://github.com/user-attachments/assets/68ee593d-d688-4074-82fe-7869b7992bd8)
如图以数字开头的句子没有被识别，请问如何解决哈？

**Comment by SWHL** [comment]
<img width="805" alt="image" src="https://github.com/user-attachments/assets/b1934c12-0634-4363-bc5e-baeb5bb3dede">

我在RapidOCR在线demo中识别，是可以识别的呀。

demo: https://huggingface.co/spaces/SWHL/RapidOCRDemo

**Reply by joseph16388** [reply]
> <img alt="image" width="805" src="https://private-user-images.githubusercontent.com/28639377/385547249-b1934c12-0634-4363-bc5e-baeb5bb3dede.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NTc3NDMsIm5iZiI6MTczMTU1NzQ0MywicGF0aCI6Ii8yODYzOTM3Ny8zODU1NDcyNDktYjE5MzRjMTItMDYzNC00MzYzLWJjNWUtYmFlYjViYjNkZWRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE0VDA0MTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmNWNiMTYwNmQ4NWM2ODJhZTNjMGE0ZGVhNmMxNGI3ODlhNGY4MDBmOTM0NmMyN2U2OTU2MThmOTE2MzgyN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4AzlxOwU3hHx0SVoAfaVvcsopfRVX6wmHpQnLQ2tYCo">
> 我在RapidOCR在线demo中识别，是可以识别的呀。
> 
> demo: https://huggingface.co/spaces/SWHL/RapidOCRDemo

我用的是[RapidVideOCR](https://github.com/SWHL/RapidVideOCR) 这个库哦

**Reply by SWHL** [reply]
我知道呀，RapidVideOCR中就调用了RapidOCR这个库的。我只是想单测一下这个图在RapidOCR中识别有么有问题。看啦是没有问题的。

**Comment by joseph16388** [comment]
![image](https://github.com/user-attachments/assets/7b385957-3eb1-48dc-b88b-88715c51cf20)
这句话，用[RapidVideOCR](https://github.com/SWHL/RapidVideOCR) 这个库识别成了INAE火箭这是放弃了比赛

如果RapidVideOCR和RapidOCR用的是完全一样的算法，那是不是VSF的锅呢

**Reply by SWHL** [reply]
可能是哈。如果想要进一步排查问题出在了哪里，需要提供VSF生成的RGBImages目录，和运行代码及环境。

**Reply by joseph16388** [reply]
98场比赛是我的一个正则表达式的错误，可以略过，主要还是下面莫名频繁多出几个英文字母的问题
样本下载地址：[e8596c9c46bc4b24842f31d0aa5f6ae5.zip](https://github.com/user-attachments/files/17760695/e8596c9c46bc4b24842f31d0aa5f6ae5.zip)

![image](https://github.com/user-attachments/assets/351abca9-5e03-4ec7-ac7f-998f3ed8a572)

![image](https://github.com/user-attachments/assets/49fc8883-46c7-42c5-bc29-2c23ece41d33)
![0_00_06_633__0_00_09_032_0000007681280019212800960](https://github.com/user-attachments/assets/faac5250-4cde-4133-9e6f-ea2369971606)

![0_00_10_533__0_00_11_699_0000007681280019212800960](https://github.com/user-attachments/assets/8e50a8cc-43d6-4c4b-a04b-abc85ac4f050)

我用在线demo试了一下，识别是正确的的。VSF的RGBImages如上也是没有问题的，但是RapidVideOCR过程中的srt文件会频繁的多出几个英语单词，如上图。

我调用的核心代码如下：
        vsf_ocr = RapidVideoSubFinderOCR(
            vsf_exe_path=vsf_exe_path,
            is_concat_rec=True,
            create_cleared_text_images=False,
            open_video_opencv=open_video_opencv,
            open_video_ffmpeg=open_video_ffmpeg,
            use_cuda=True,
            top_video_image_percent_end=top_video_image_percent_end,
            bottom_video_image_percent_end=bottom_video_image_percent_end,
            left_video_image_percent_end=left_video_image_percent_end,
            right_video_image_percent_end=right_video_image_percent_end,
            text_score=0.4
        )

版本是rapid-videocr==2.2.14，应该是最新版
运行环境WIN11,Python3.11.5的虚拟环境

有劳指出错误了！

**Reply by SWHL** [reply]
INAE这几个单词我猜应该是下面红框的单词。建议字幕的框宽度搞得再小一点。
<img width="830" alt="image" src="https://github.com/user-attachments/assets/4229a12b-deea-46ff-b2ec-0961346aac01">

我这里也看看有没有更好的办法过滤这个。

**Reply by joseph16388** [reply]
> INAE这几个单词我猜应该是下面红框的单词。建议字幕的框宽度搞得再小一点。 <img alt="image" width="830" src="https://private-user-images.githubusercontent.com/28639377/386443502-4229a12b-deea-46ff-b2ec-0961346aac01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NDUwMDMsIm5iZiI6MTczMTY0NDcwMywicGF0aCI6Ii8yODYzOTM3Ny8zODY0NDM1MDItNDIyOWExMmItZGVlYS00NmZmLWIyZWMtMDk2MTM0NmFhYzAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE1VDA0MjUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwNTY5YmRiYTNlNWZkNGI3YjI3YzY5NjA4NTNmOWEzYTQyZDk3MDBkNDdmYWIxOTJkMmE2YjAyNzJiMThiYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7Z025Hzfg-wgdJvjDJutBTAyumpSYFMvBA7qQucz3BU">
> 
> 我这里也看看有没有更好的办法过滤这个。

对的，不过这个框是针对整个视频的，如果能智能识别出仅字幕区域就好了。
1、比如指定字幕的颜色进行过滤这种(也就是在OCR解析素材之前只提取与字幕颜色相关的部分)
2、或者OCR本身支持只识别特定颜色范围的字幕
感觉这个功能超级实用，您看有什么设定可以做到这个哈？

**Reply by joseph16388** [reply]
> INAE这几个单词我猜应该是下面红框的单词。建议字幕的框宽度搞得再小一点。 <img alt="image" width="830" src="https://private-user-images.githubusercontent.com/28639377/386443502-4229a12b-deea-46ff-b2ec-0961346aac01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NTkwOTMsIm5iZiI6MTczMTY1ODc5MywicGF0aCI6Ii8yODYzOTM3Ny8zODY0NDM1MDItNDIyOWExMmItZGVlYS00NmZmLWIyZWMtMDk2MTM0NmFhYzAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE1VDA4MTk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyZTZkOTA1ZmVjYzg5MDczMDRjMWUwNDc3Y2Y5MGU0ODU0NDA2ZDFmYzBjYTU5NDcwZDE3M2U2MzE0M2JlMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1UBAnY8fcUk3QdJMe9LVBuCnQv96LvaJZrvlXCoFJxA">
> 
> 我这里也看看有没有更好的办法过滤这个。

还有一点就是，在线demo的这个并没有去识别衣服上的字母，但VideOCR却去识别了？我理解应该是有些参数不同导致的吧，既然在线demo可以做到，那是不是调整下参数就行了?
![image](https://github.com/user-attachments/assets/82349f08-121b-4bf4-b327-4195c0c5bbee)

**Reply by SWHL** [reply]
> 对的，不过这个框是针对整个视频的，如果能智能识别出仅字幕区域就好了。
> 1、比如指定字幕的颜色进行过滤这种(也就是在OCR解析素材之前只提取与字幕颜色相关的部分)
> 2、或者OCR本身支持只识别特定颜色范围的字幕

根据字幕颜色不太可行。还是提前框选一下比较靠谱。因为视频背景是一直在变化的。如果字幕是黑底的，那可以。

**Comment by joseph16388** [comment]
还有如果可以通过API得到每个文字的BoundingBox区域信息，那应该可以有助于后处理的

**Comment by SWHL** [comment]
这个再支持了，等两天吧

**Reply by joseph16388** [reply]
> 这个再支持了，等两天吧

期待！

**Reply by SWHL** [reply]
重新安装一下，应该会安装`rapidocr_onnxruntime==1.4.0`，这个版本是支持的。详细参见：https://github.com/RapidAI/RapidOCR/releases/tag/v1.4.0

**Reply by joseph16388** [reply]
> 重新安装一下，应该会安装`rapidocr_onnxruntime==1.4.0`，这个版本是支持的。详细参见：https://github.com/RapidAI/RapidOCR/releases/tag/v1.4.0

收到！

**Reply by joseph16388** [reply]
> 重新安装一下，应该会安装`rapidocr_onnxruntime==1.4.0`，这个版本是支持的。详细参见：https://github.com/RapidAI/RapidOCR/releases/tag/v1.4.0
还有请教下我直接用RapidOCR()为什么INAE这几个英文字母就不会去识别了？但用VideOCR就会去识别呢？

**Reply by SWHL** [reply]
等我复现看看

**Comment by SWHL** [comment]
![image](https://github.com/user-attachments/assets/aa77f2ce-4b9d-4aae-b333-6b3ec6e4926d)
这是我在Google Colab上跑的这几个例子，并没有出现你这里情况。我猜测和操作系统有关系。之前遇到过，macOS上就是会少检测几个框。

**Reply by joseph16388** [reply]
我用的是WIN11系统，INAE这个框是检测出来了。我的需求主要是1、确定一共有几个框 2、提取框内主要的颜色值3、根据字幕的颜色值，过滤掉不是字幕的几个框 4、这些步骤能在srt产出之前进行灵活的配置以达到过滤的效果哈

**Reply by SWHL** [reply]
你这需求需要你这边自己去探索了，太细了，目前不是我能解决的。欢迎提PR哈

**Reply by joseph16388** [reply]
> 你这需求需要你这边自己去探索了，太细了，目前不是我能解决的。欢迎提PR哈

嗯，目前只是希望有个多矩形框的boundingbox范围的返回值，然后拿来后处理，哈哈。。
